packages/manager/apps/pci-databases-analytics/src/lib/durationHelper.ts

```diff
@@ -9,30 +9,57 @@ export function durationStringToHuman(durationString: string, locale: Locale) {
 }
 
 export function durationStringToDuration(durationString: string) {
-  // Regular expression to match the duration parts
-  const pattern = /(\d+Y)?(\d+M)?(\d+D)?(\d+H)?(\d+m)?(\d+S)?/;
-  const matches = pattern.exec(durationString);
-
-  // Convert matched values to numbers, defaulting to 0 if not present
-  const years = matches[1] ? parseInt(matches[1], 10) : 0;
-  const months = matches[2] ? parseInt(matches[2], 10) : 0;
-  const days = matches[3] ? parseInt(matches[3], 10) : 0;
-  const hours = matches[4] ? parseInt(matches[4], 10) : 0;
-  const minutes = matches[5] ? parseInt(matches[5], 10) : 0;
-  const seconds = matches[6] ? parseInt(matches[6], 10) : 0;
-  // Construct and return the duration object
-  return { years, months, days, hours, minutes, seconds };
+  const pattern = /(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(\d+H)?(\d+m)?(\d+S)?/g;
+  const matches = [...durationString.matchAll(pattern)];
+
+  const {
+    years,
+    months,
+    weeks,
+    days,
+    hours,
+    minutes,
+    seconds,
+  } = matches.reduce(
+    (acc, match) => {
+      if (match[1]) acc.years += parseInt(match[1], 10);
+      if (match[2]) acc.months += parseInt(match[2], 10);
+      if (match[3]) acc.weeks += parseInt(match[3], 10);
+      if (match[4]) acc.days += parseInt(match[4], 10);
+      if (match[5]) acc.hours += parseInt(match[5], 10);
+      if (match[6]) acc.minutes += parseInt(match[6], 10);
+      if (match[7]) acc.seconds += parseInt(match[7], 10);
+      return acc;
+    },
+    {
+      years: 0,
+      months: 0,
+      weeks: 0,
+      days: 0,
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+    },
+  );
+
+  // Adjust total days based on weeks
+  const totalDays = days + weeks * 7;
+
+  return { years, months, days: totalDays, hours, minutes, seconds };
```

Comment:
Why don't use :
https://date-fns.org/v4.1.0/docs/formatDuration
https://date-fns.org/v4.1.0/docs/formatISODuration

?

Reply:
The durationString parameter is not exactly a ISO 8601 string, so we had to handle the conversion ourselves here

Reply:
Ah no I understand this usecase. So I approve